107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.jsr107;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.CacheConfigurationBuilder;
+import org.ehcache.config.loaderwriter.DefaultCacheLoaderWriterConfiguration;
+import org.ehcache.config.xml.XmlConfiguration;
+import org.ehcache.internal.store.heap.service.OnHeapStoreServiceConfiguration;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.slf4j.Logger;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.cache.configuration.CacheEntryListenerConfiguration;
+import javax.cache.configuration.CompleteConfiguration;
+import javax.cache.configuration.Configuration;
+import javax.cache.configuration.Factory;
+import javax.cache.integration.CacheLoader;
+import javax.cache.integration.CacheWriter;
+
+import static org.ehcache.config.CacheConfigurationBuilder.newCacheConfigurationBuilder;
+
+/**
+ * ConfigurationMerger
+ */
+class ConfigurationMerger {
+
+  private final XmlConfiguration xmlConfiguration;
+  private final Jsr107Service jsr107Service;
+  private final Eh107CacheLoaderWriterFactory cacheLoaderWriterFactory;
+  private final Logger log;
+
+  ConfigurationMerger(org.ehcache.config.Configuration ehConfig, Jsr107Service jsr107Service, Eh107CacheLoaderWriterFactory cacheLoaderWriterFactory, Logger log) {
+    if (ehConfig instanceof XmlConfiguration) {
+      xmlConfiguration = (XmlConfiguration) ehConfig;
+    } else {
+      xmlConfiguration = null;
+    }
+    this.jsr107Service = jsr107Service;
+    this.cacheLoaderWriterFactory = cacheLoaderWriterFactory;
+    this.log = log;
+  }
+
+  <K, V> ConfigHolder<K, V> mergeConfigurations(String cacheName, Configuration<K, V> configuration) {
+    final Eh107CompleteConfiguration<K, V> jsr107Configuration = new Eh107CompleteConfiguration<K, V>(configuration);
+
+    Eh107Expiry<K, V> expiryPolicy = null;
+    CacheLoaderWriter<? super K, V> loaderWriter = null;
+    try {
+      CacheConfigurationBuilder<K, V> builder = newCacheConfigurationBuilder();
+      String templateName = jsr107Service.getTemplateNameForCache(cacheName);
```

Comment:
We only build off this templateName if the xmlConfiguration is non null... could we move this assignment (and declaration) inside the `if(xmlConfiguration != null)` conditional?  This causes cascading issues with the logging further down... but if the conditional on those logging calls are aligned with the conditional on the creation of the builder then thats not immediately obvious.  It's also not clear to me that they aren't either brittle now, or candidates for becoming brittle later.